pkg/services/provisioning/dashboards/file_reader.go

```diff
@@ -51,34 +51,25 @@ func NewDashboardFileReader(cfg *DashboardsAsConfig, log log.Logger) (*fileReade
 	}, nil
 }
 
-func (fr *fileReader) ReadAndListen(ctx context.Context) error {
-	if err := fr.startWalkingDisk(); err != nil {
-		fr.log.Error("failed to search for dashboards", "error", err)
-	}
-
-	ticker := time.NewTicker(time.Duration(int64(time.Second) * fr.Cfg.UpdateIntervalSeconds))
-
-	running := false
-
+// pollChanges periodically runs startWalkingDisk based on interval specified in the config.
+func (fr *fileReader) pollChanges(ctx context.Context) {
+	ticker := time.Tick(time.Duration(int64(time.Second) * fr.Cfg.UpdateIntervalSeconds))
 	for {
 		select {
-		case <-ticker.C:
-			if !running { // avoid walking the filesystem in parallel. in-case fs is very slow.
-				running = true
-				go func() {
-					if err := fr.startWalkingDisk(); err != nil {
-						fr.log.Error("failed to search for dashboards", "error", err)
-					}
-					running = false
-				}()
+		case <-ticker:
+			if err := fr.startWalkingDisk(); err != nil {
```

Comment:
I removed the `running` check from here. Based on docs:
> // It adjusts the intervals or drops ticks to make up for slow receivers.

> 	// Give the channel a 1-element time buffer.
	// If the client falls behind while reading, we drop ticks
	// on the floor until the client catches up.

it should not be necessary.

Reply:
Think this makes sense, just trying to understand. Is it correctly understood that since we no longer use a go routine we simulate a 1-element time buffer?

Reply:
Well the 1-element time buffer is already implemented in the Ticker. So before we had a goroutine here so we could skip ticks in case we were still walking disc, so as to not block the ticker I assume. But the Ticker seems to be doing that by itself so we don't have to. So right now if `startWalkingDisk` takes too long Ticker will see that the consumer is blocked and won't push a value into the channel skipping a tick (or more). So the functionality should be exactly the same as before.

Reply:
Great find! :) Love this improvement.